packages/api/src/command/medical/patient/settings/create-patient-settings.ts

```diff
@@ -99,32 +106,48 @@ export async function upsertPatientSettingsForPatientList({
 }
 
 /**
- * Processes a batch of patient settings updates
+ * Processes a batch of patient settings upserts with retries
  */
 async function processBatch({
-  patients,
+  patientIds,
   cxId,
   settings,
+  log,
 }: {
-  patients: string[];
+  patientIds: string[];
   cxId: string;
   settings: PatientSettingsData;
+  log: typeof console.log;
 }): Promise<BatchProcessingResult> {
-  try {
-    const processedCount = await upsertPatientSettings({
-      patientIds: patients,
-      cxId,
-      settings,
-    });
-    return { processedCount, failedCount: 0 };
-  } catch (error) {
-    return { processedCount: 0, failedCount: patients.length, error: error as Error };
+  for (let attempt = 1; attempt <= MAX_BATCH_RETRIES; attempt++) {
+    try {
+      const processedCount = await upsertPatientSettings({
+        patientIds,
+        cxId,
+        settings,
+      });
+      return { processedCount, failedCount: 0 };
+    } catch (error) {
+      const isLastAttempt = attempt === MAX_BATCH_RETRIES;
+
+      if (!isLastAttempt) {
+        log(
+          `Batch operation failed, attempt ${attempt}/${MAX_BATCH_RETRIES}: ${errorToString(error)}`
+        );
+      }
```

Comment:
NABD, but since we're logging this, might as well add a log for the last attempt, saying it error'd and we're not trying again.